internal/storage/db.go

```diff
@@ -76,3 +81,43 @@ func (s *DBStorage) AssignTokens(ctx context.Context, issuanceWeek, firstAutomat
 	_, err = s.DBS.DBS().Writer.ExecContext(ctx, assignTokensQuery, issuanceWeek, weekLimit.String())
 	return err
 }
+
+var tokensPerWeekQuery = `
+SELECT 
```

Comment:
This query is so spooky to me but I think it's fine, let's try it.

I think either accept/reject my comments on API shape and let's ship it on dev

Reply:
I liked all the suggestions ran through some tests with the dump you gave me and things appear to be working.